### ISSUE TITLE
Unable to reach Sonar server via https

### ISSUE THREAD

**Issue by MatveyNeyman** [bellingard/sonar-scanner-npm#35]
We use our own certificates to connect to server.

Following error appears:
ERROR: SonarQube server [https://xxxx.xxx.xx] can not be reached

Everything works fine if I use Java-based Sonar Scanner (sonar-scanner-3.0.3.778-macosx) with custom sonar-scanner-3.0.3.778-macosx/jre/lib/security/cacerts file.

Tried to copy same cacerts file to JRE's security folder but no luck.

**Comment by muhlba91** [comment]
Are there any updates on this?

**Comment by bellingard** [comment]
I acknowledge this limitation. Still, I won't spend time on this for now given the fact that few people faced this limitation (compared to the monthly downloads of ~ 20k+).

**Comment by zpavic** [comment]
Running into this issue right now for our organization. We've had the host JDK/JRE cacerts files updated but it was a bit surprising to find another JRE embedded. 

Do we have an option to use the system Java installation?

**Comment by muhlba91** [comment]
@zpavic - yep, i didn't find much of a workaround to that.
The only way I found was calling the `scanCLI` method method and passing the `-Djavax.net.ssl.keyStore` and `-Djavax.net.ssl.trustStore` variable with the system Java's cacert path in.

However, in my particular case the issue was more the allowed ciphers. A usual Java installation doesn't contain many ciphers and if your HTTPS proxy/server/... does not allow at least one of the supported ones, this exception gets thrown as well.

**Comment by bellingard** [comment]
@zpavic or @muhlba91 , have you tried passing the location of the keystore using the `$SONAR_SCANNER_OPTS` environment variable? Something like:

`export SONAR_SCANNER_OPTS="-Djavax.net.ssl.trustStore=/paht/to/my-cacerts -Djavax.net.ssl.keyStore=/path/to/my-keystore-file"`

**Comment by muhlba91** [comment]
@bellingard yep. that also works fine. :)

**Comment by bellingard** [comment]
@muhlba91 So there's no issue?

**Comment by muhlba91** [comment]
@bellingard - the issue for me was that Java doesn‘t include more secure ciphers for https connections. After adding a lower bit sized cipher to my web server‘s config it worked. :)
However, in general regarding the certificate issue and using an external cacert store, you can either pass the command line arguments directly to the scanCLI method or export the SONAR_SCANNER_OPTS property as you mentioned. Both works fine as long as the cacert store contains the correct certificate.

**Comment by zpavic** [comment]
Setting the `SONAR_SCANNER_OPTS` worked for me. Thanks!

**Comment by MMirabito** [comment]
Drear All,

I am having bizarre behavior using Sonar CLI. Today we updated to **SonarQube 7.6** and sonar-**scanner-cli-3.3.0.1492** hosted on a windows server. SonarQube Server is fronted by apache with a self-signed cert and its CA is in the trust store. When I try to run the CLI (_**v. 3.3.0.1492**_) within the command line of the server (**sonar-scanner -X**) I receive the following error:

`
11:29:53.620 ERROR: Error during SonarQube Scanner execution
org.sonarsource.scanner.api.internal.ScannerException: Unable to execute SonarQube
        at org.sonarsource.scanner.api.internal.IsolatedLauncherFactory.lambda$createLauncher$0(IsolatedLauncherFactory.
java:85)
        at java.security.AccessController.doPrivileged(Native Method)
        at org.sonarsource.scanner.api.internal.IsolatedLauncherFactory.createLauncher(IsolatedLauncherFactory.java:74)
        at org.sointernal.IsolatedLauncherFactory.createLauncher(IsolatedLauncherFactory.java:70)
        at org.sonarsource.scanner.api.EmbeddedScanner.doStart(EmbeddedScanner.java:181)
        at org.sonarsource.scanner.api.EmbeddedScanner.start(EmbeddedScanner.java:122)
        at org.sonarsource.scanner.cli.Main.execute(Main.java:73)
        at org.sonarsource.scanner.cli.Main.main(Main.java:61)
Caused by: java.lang.IllegalStateException: Fail to get bootstrap index from server
        at org.sonarsource.scanner.api.internal.BootstrapIndexDownloader.getIndex(BootstrapIndexDownloader.java:42)
        at org.sonarsource.scanner.api.internal.JarDownloader.getScannerEngineFiles(JarDownloader.java:58)
        at org.sonarsource.scanner.api.internal.JarDownloader.download(JarDownloader.java:53)
        at org.sonarsource.scanner.api.internal.IsolatedLauncherFactory.lambda$createLauncher$0(IsolatedLauncherFactory.
java:76)
        ... 7 more
Caused by: javax.net.ssl.SSLPeerUnverifiedException: Hostname aspv-nifm-alm.cdc.gov not verified:
    certificate: sha256/l7poD2QiqENbsMFsh2AGmrdy2BXmhUj9G9N8T+cq8dw=
    DN: EMAILADDRESS=xoxo@xox.com, CN=xxx-xxx-xxx.xox.gov, OU=XOXO, O=XOX, L=Xxxxx, ST=XX, C=ZZ
    subjectAltNames: []
        at org.sonarsource.scanner.api.internal.shaded.okhttp.internal.connection.RealConnection.connectTls(RealConnecti
on.java:329)
        at org.sonarsource.scanner.api.internal.shaded.okhttp.internal.connection.RealConnection.establishProtocol(RealC
onnection.java:282)
        at org.sonarsource.scanner.api.internal.shaded.okhttp.internal.connection.RealConnection.connect(RealConnection.
java:167)
        at org.sonarsource.scanner.api.internal.shaded.okhttp.internal.connection.StreamAllocation.findConnection(Stream
Allocation.java:257)
`

I have tried the suggestions above but nothing appears to be working. What is really strange is if I run the CLI **_sonar-scanner-cli-3.2.0.1227_** I do not have the issue.
Does anyone know what could have changed between versions that is causing the issue and what I can do to resolve it?

Thanks in advance for any suggestions

max

**Comment by bellingard** [comment]
@henryju Does that ring a bell to you? Could it be a regression on our side.

**Comment by henryju** [comment]
The only change I'm thinking of is https://jira.sonarsource.com/browse/SCANNERAPI-179 (update of okhttp library). But this is hard to investigate "remotely". @MMirabito I suggest you look at https://docs.oracle.com/javase/8/docs/technotes/guides/security/jsse/ReadDebug.html to know how to generate (very) verbose logs about SSL handshake to see what's going on.

**Comment by MMirabito** [comment]
@henryju Thanks for the response. I debugged the SSL connections as follows:

```
SONAR_SCANNER_OPTS=-Djavax.net.debug=all
sonar-scanner –X
```
Below is the partial SSL handshake. I placed the output of  v 3.3.0.1492 and v 3.2.0.1227 side by side everything appears to be the same until right before  v 3.3.0.1492 craps out. Any ideas where the root cause might be? IS there a way to pass an argument and let the CLI and dependent libraries to skip SSL verification just for testing purposes?

Thanks again for any suggestions
max

Partial SSL handshake debug extract 
### **v 3.3.0.1492 - BAD SSL Connection**
*** Finished
verify_data:  { 71, 177, 243, 29, 124, 31, 146, 167, 0, 234, 84, 214 }
**
%% Cached client session: [Session-1, TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256]
[read] MD5 and SHA1 hashes:  len = 16
0000: 14 00 00 0C 47 B1 F3 1D   7C 1F 92 A7 00 EA 54 D6  ....G.........T.
main, called close()
main, called closeInternal(true)
**main, SEND TLSv1.2 ALERT:  warning, description = close_notify
Padded plaintext before ENCRYPTION:  len = 2
0000: 01 00                                              ..
main, WRITE: TLSv1.2 Alert, length = 26
[Raw write]: length = 31
0000: 15 03 03 00 1A 00 00 00   00 00 00 00 01 2D 84 C8  .............-..
0010: 15 A3 2A 04 2F 2B D4 7D   1F AD 2E 26 55 98 A2     ..*./+.....&U..
main, called closeSocket(true)**
15:53:54.590 INFO: ------------------------------------------------------------------------
15:53:54.591 INFO: EXECUTION FAILURE
15:53:54.591 INFO: ------------------------------------------------------------------------
15:53:54.591 INFO: Total time: 1.472s
15:53:54.678 INFO: Final Memory: 7M/307M
15:53:54.679 INFO: ------------------------------------------------------------------------
Finalizer, called close()
Finalizer, called closeInternal(true)

### **v 3.2.0.1227 - GOOD SSL Connection**
*** Finished
verify_data:  { 141, 173, 140, 206, 205, 52, 116, 162, 49, 113, 172, 209 }
**
%% Cached client session: [Session-2, TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256]
[read] MD5 and SHA1 hashes:  len = 16
0000: 14 00 00 0C 8D AD 8C CE   CD 34 74 A2 31 71 AC D1  .........4t.1q..
main, setSoTimeout(60000) called
Padded plaintext before ENCRYPTION:  len = 216
0000: 47 45 54 20 2F 53 4F 4E   41 52 51 55 42 45 2F 61  GET /SONARQUBE/a
0010: 70 69 2F 73 65 74 74 69   6E 67 73 2F 76 61 6C 75  pi/settings/valu
0020: 65 73 2E 70 72 6F 74 6F   62 75 66 20 48 54 54 50  es.protobuf HTTP
0030: 2F 31 2E 31 0D 0A 41 63   63 65 70 74 3A 20 61 70  /1.1..Accept: ap
0040: 70 6C 69 63 61 74 69 6F   6E 2F 6A 73 6F 6E 0D 0A  plication/json..
0050: 41 63 63 65 70 74 2D 43   68 61 72 73 65 74 3A 20  Accept-Charset: 
0060: 55 54 46 2D 38 0D 0A 48   6F 73 74 3A 20 61 73 70  UTF-8..Host: xxx
0070: 76 2D 6E 69 66 6D 2D 61   6C 6D 2E 63 64 63 2E 67  v-nifm-xxx.xxx.c
0080: 6F 76 0D 0A 43 6F 6E 6E   65 63 74 69 6F 6E 3A 20  com..Connection: 
0090: 4B 65 65 70 2D 41 6C 69   76 65 0D 0A 41 63 63 65  Keep-Alive..Acce
00A0: 70 74 2D 45 6E 63 6F 64   69 6E 67 3A 20 67 7A 69  pt-Encoding: gzi
00B0: 70 0D 0A 55 73 65 72 2D   41 67 65 6E 74 3A 20 53  p..User-Agent: S
00C0: 63 61 6E 6E 65 72 43 6C   69 2F 33 2E 32 2E 30 2E  cannerCli/3.2.0.
00D0: 31 32 32 37 0D 0A 0D 0A                            1227....
main, WRITE: TLSv1.2 Application Data, length = 240
[Raw write]: length = 245

**Comment by henryju** [comment]
@MMirabito I'm not an SSL expert, so hard to say what's wrong. I have prepared a new scanner version, including the latest version of okhttp library, even if I haven't found anything particular in the release notes:
https://repox.sonarsource.com/sonarsource/org/sonarsource/scanner/cli/sonar-scanner-cli/3.3.0.1532/sonar-scanner-cli-3.3.0.1532.zip

Could you please try and report if it fixes your issue?

**Comment by MMirabito** [comment]
@henryju I just tried  but the same issue continues to occur - this really bizarre especially because I do not understand the error so I am having a hard time pinning down the root cause. Below is a partial debug extract. Any other suggestions?

**NOTE:** The ZIP file I downloaded did not include the JRE so when I run the CLI it's picking up the JRE of the the server.  

Thanks,
max


```
D:\MyPrograms\sonar-scanner-3.3.0.1532\bin>sonar-scanner -X
05:41:23.012 INFO: Scanner configuration file: D:\MyPrograms\sonar-scanner-3.3.0.1532\bin\..\conf\sonar-scanner.
properties
05:41:23.024 INFO: Project root configuration file: NONE
05:41:23.074 INFO: SonarQube Scanner 3.3.0.1532
05:41:23.075 INFO: Java 1.8.0_191 Oracle Corporation (64-bit)
05:41:23.075 INFO: Windows Server 2012 R2 6.3 amd64
05:41:23.076 INFO: SONAR_SCANNER_OPTS=-Djavax.net.debug=all
trustStore is: D:\MyPrograms\Java\jdk1.8.0\jre\lib\security\cacerts
trustStore type is : jks
trustStore provider is :
init truststore
```




```
*** Finished
verify_data:  { 47, 254, 253, 239, 205, 55, 167, 171, 197, 121, 148, 247 }
***
update handshake state: finished[20]
upcoming handshake states: server change_cipher_spec[-1]
upcoming handshake states: server finished[20]
[write] MD5 and SHA1 hashes:  len = 16
0000: 14 00 00 0C 2F FE FD EF   CD 37 A7 AB C5 79 94 F7  ..../....7...y..
Padded plaintext before ENCRYPTION:  len = 16
0000: 14 00 00 0C 2F FE FD EF   CD 37 A7 AB C5 79 94 F7  ..../....7...y..
main, WRITE: TLSv1.2 Handshake, length = 40
[Raw write]: length = 45
0000: 16 03 03 00 28 00 00 00   00 00 00 00 00 60 99 6E  ....(........`.n
0010: 23 52 88 1C 0F D4 55 65   0B 90 2F D7 1D BA 7F F9  #R....Ue../.....
0020: 41 99 63 49 88 26 3E ED   2A D2 DF E3 BC           A.cI.&>.*....
[Raw read]: length = 5
0000: 14 03 03 00 01                                     .....
[Raw read]: length = 1
0000: 01                                                 .
main, READ: TLSv1.2 Change Cipher Spec, length = 1
update handshake state: change_cipher_spec
upcoming handshake states: server finished[20]
[Raw read]: length = 5
0000: 16 03 03 00 28                                     ....(
[Raw read]: length = 40
0000: D9 F6 8A F9 CF CA 47 53   B2 9F 59 A2 5B C4 C3 81  ......GS..Y.[...
0010: 90 5A CA 84 B3 CE BA 36   C9 45 A8 3C BA E6 22 8A  .Z.....6.E.<..".
0020: 8D FF AB 45 4F A5 A8 CC                            ...EO...
main, READ: TLSv1.2 Handshake, length = 40
Padded plaintext after DECRYPTION:  len = 16
0000: 14 00 00 0C 42 DF FF 30   6D A6 71 06 6D 61 D3 3B  ....B..0m.q.ma.;
check handshake state: finished[20]
update handshake state: finished[20]
*** Finished
verify_data:  { 66, 223, 255, 48, 109, 166, 113, 6, 109, 97, 211, 59 }
***
%% Cached client session: [Session-1, TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384]
[read] MD5 and SHA1 hashes:  len = 16
0000: 14 00 00 0C 42 DF FF 30   6D A6 71 06 6D 61 D3 3B  ....B..0m.q.ma.;
main, called close()
main, called closeInternal(true)
main, SEND TLSv1.2 ALERT:  warning, description = close_notify
Padded plaintext before ENCRYPTION:  len = 2
0000: 01 00                                              ..
main, WRITE: TLSv1.2 Alert, length = 26
[Raw write]: length = 31
0000: 15 03 03 00 1A 00 00 00   00 00 00 00 01 12 C0 9D  ................
0010: 11 2F CD 6A 54 1F 6D B3   D3 41 82 18 D4 AE 6F     ./.jT.m..A....o
main, called closeSocket(true)
05:41:27.990 ERROR: SonarQube server [https://xxxx-nifm-xxx.aaa.com/SONARQUBE] can not be reached
05:41:27.993 INFO: ------------------------------------------------------------------------
05:41:27.994 INFO: EXECUTION FAILURE
05:41:27.995 INFO: ------------------------------------------------------------------------
05:41:27.997 INFO: Total time: 5.033s
05:41:28.085 INFO: Final Memory: 7M/307M
Finalizer, called close()
Finalizer, called closeInternal(true)
05:41:28.111 INFO: ------------------------------------------------------------------------
05:41:28.115 ERROR: Error during SonarQube Scanner execution
org.sonarsource.scanner.api.internal.ScannerException: Unable to execute SonarQube
        at org.sonarsource.scanner.api.internal.IsolatedLauncherFactory.lambda$createLauncher$0(IsolatedLauncherFactory.
java:85)
        at java.security.AccessController.doPrivileged(Native Method)
        at org.sonarsource.scanner.api.internal.IsolatedLauncherFactory.createLauncher(IsolatedLauncherFactory.java:74)
        at org.sonarsource.scanner.api.internal.IsolatedLauncherFactory.createLauncher(IsolatedLauncherFactory.java:70)
        at org.sonarsource.scanner.api.EmbeddedScanner.doStart(EmbeddedScanner.java:181)
        at org.sonarsource.scanner.api.EmbeddedScanner.start(EmbeddedScanner.java:122)
        at org.sonarsource.scanner.cli.Main.execute(Main.java:73)
        at org.sonarsource.scanner.cli.Main.main(Main.java:61)
Caused by: java.lang.IllegalStateException: Fail to get bootstrap index from server
        at org.sonarsource.scanner.api.internal.BootstrapIndexDownloader.getIndex(BootstrapIndexDownloader.java:42)
        at org.sonarsource.scanner.api.internal.JarDownloader.getScannerEngineFiles(JarDownloader.java:58)
        at org.sonarsource.scanner.api.internal.JarDownloader.download(JarDownloader.java:53)
        at org.sonarsource.scanner.api.internal.IsolatedLauncherFactory.lambda$createLauncher$0(IsolatedLauncherFactory.
java:76)
        ... 7 more
Caused by: javax.net.ssl.SSLPeerUnverifiedException: Hostname xxxx-nifm-xxx.aaa.com not verified:
    certificate: sha256/l7poD2QiqENbsMFsh2AGmrdy2BXmhUj9G9N8T+cq8dw=
    DN: EMAILADDRESS=asd@aaa.com, CN=xxxx-nifm-xxx.aaa.com, OU=XXXXXX, O=XXX, L=Xxxxxx, ST=XX, C=XX
    subjectAltNames: []
        at org.sonarsource.scanner.api.internal.shaded.okhttp.internal.connection.RealConnection.connectTls(RealConnecti
on.java:334)
        at org.sonarsource.scanner.api.internal.shaded.okhttp.internal.connection.RealConnection.establishProtocol(RealC
onnection.java:284)
        at org.sonarsource.scanner.api.internal.shaded.okhttp.internal.connection.RealConnection.connect(RealConnection.
java:169)
        at org.sonarsource.scanner.api.internal.shaded.okhttp.internal.connection.StreamAllocation.findConnection(Stream
Allocation.java:257)
        at org.sonarsource.scanner.api.internal.shaded.okhttp.internal.connection.StreamAllocation.findHealthyConnection
(StreamAllocation.java:135)
        at org.sonarsource.scanner.api.internal.shaded.okhttp.internal.connection.StreamAllocation.newStream(StreamAlloc
ation.java:114)
        at org.sonarsource.scanner.api.internal.shaded.okhttp.internal.connection.ConnectInterceptor.intercept(ConnectIn
terceptor.java:42)
        at org.sonarsource.scanner.api.internal.shaded.okhttp.internal.http.RealInterceptorChain.proceed(RealInterceptor
Chain.java:147)
        at org.sonarsource.scanner.api.internal.shaded.okhttp.internal.http.RealInterceptorChain.proceed(RealInterceptor
Chain.java:121)
        at org.sonarsource.scanner.api.internal.shaded.okhttp.internal.cache.CacheInterceptor.intercept(CacheInterceptor
.java:94)
        at org.sonarsource.scanner.api.internal.shaded.okhttp.internal.http.RealInterceptorChain.proceed(RealInterceptor
Chain.java:147)
        at org.sonarsource.scanner.api.internal.shaded.okhttp.internal.http.RealInterceptorChain.proceed(RealInterceptor
Chain.java:121)
        at org.sonarsource.scanner.api.internal.shaded.okhttp.internal.http.BridgeInterceptor.intercept(BridgeIntercepto
r.java:93)
        at org.sonarsource.scanner.api.internal.shaded.okhttp.internal.http.RealInterceptorChain.proceed(RealInterceptor
Chain.java:147)
        at org.sonarsource.scanner.api.internal.shaded.okhttp.internal.http.RetryAndFollowUpInterceptor.intercept(RetryA
ndFollowUpInterceptor.java:125)
        at org.sonarsource.scanner.api.internal.shaded.okhttp.internal.http.RealInterceptorChain.proceed(RealInterceptor
Chain.java:147)
        at org.sonarsource.scanner.api.internal.shaded.okhttp.internal.http.RealInterceptorChain.proceed(RealInterceptor
Chain.java:121)
        at org.sonarsource.scanner.api.internal.shaded.okhttp.RealCall.getResponseWithInterceptorChain(RealCall.java:264
)
        at org.sonarsource.scanner.api.internal.shaded.okhttp.RealCall.execute(RealCall.java:93)
        at org.sonarsource.scanner.api.internal.ServerConnection.callUrl(ServerConnection.java:113)
        at org.sonarsource.scanner.api.internal.ServerConnection.downloadString(ServerConnection.java:98)
        at org.sonarsource.scanner.api.internal.BootstrapIndexDownloader.getIndex(BootstrapIndexDownloader.java:39)
        ... 10 more
```

**Comment by henryju** [comment]
The link I sent to you is indeed not embedding a JRE. You'll have to import your self signed certificate in the truststore of the JVM that is actually used.
Alternatively, you can download a version with a JRE, like:
https://repox.sonarsource.com/sonarsource/org/sonarsource/scanner/cli/sonar-scanner-cli/3.3.0.1532/sonar-scanner-cli-3.3.0.1532-windows.zip
But also don't forget to import your certificate.

**Comment by MMirabito** [comment]
@henryju let me rephrase my statement the cert and chain/CA have been imported in both the JREs packaged in the CLI and the JRE version that is installed on the server. In either case I get the same problem.

**Comment by henryju** [comment]
Random idea: looking at your certificate and the way okhttp is [checking hostname](https://github.com/square/okhttp/blob/933bb401e8d70b4566ea0a7c6f1e3b9817ff1979/okhttp/src/main/java/okhttp3/internal/tls/OkHostnameVerifier.java#L75), I think you should try to set the hostname in the `subjectAltNames` field:

In your case it is empty:

> certificate: sha256/l7poD2QiqENbsMFsh2AGmrdy2BXmhUj9G9N8T+cq8dw=
>     DN: EMAILADDRESS=asd@aaa.com, CN=xxxx-nifm-xxx.aaa.com, OU=XXXXXX, O=XXX, L=Xxxxxx, ST=XX, C=XX
>     subjectAltNames: []

While on sonarcloud.io, we have:

> SubjectAlternativeName [
>           DNSName: *.sonarcloud.io
>           DNSName: sonarcloud.io
>         ]

**Comment by henryju** [comment]
And this commit seems to prove my point:
https://github.com/square/okhttp/commit/cd84d79c3574de81f23802f832f2ead6ad38a735

Previously, having the hostname in the CN was enough. Not anymore.

**Comment by MMirabito** [comment]
You were reading my mind we started looking into that as well  - we are in the process of having the cert regenerated with a valid SubjectAlternativeName. I will keep you posted of what happens, but as you pointed from the okhttp commit they are not as forgiving with their new version  so we need to fix the cert.

Thanks again 
max

**Comment by MMirabito** [comment]
@henryju a quick note to let you know that after we deployed the new cert with the SubjectAlternativeName everything is now working. 

Should we continue to use 3.3.0.1492 or the latest build 3.3.0.1532?

Thanks again for the help and my apologies for having taxed your time  on an issue that was clearly on our side.

Thanks
max

**Comment by gfvirga** [comment]
> 
> 
> And this commit seems to prove my point:
> [square/okhttp@cd84d79](https://github.com/square/okhttp/commit/cd84d79c3574de81f23802f832f2ead6ad38a735)
> 
> Previously, having the hostname in the CN was enough. Not anymore.

Hello,

Do you know what version of the scanner it started to be a requirement for SubjectAlternativeName to have entries?

**Comment by MMirabito** [comment]
@gfvirga

Not sure if this question if for me  or for @henryju but for what is worth we did not see an issue on our side until we moved to 3.3.0.1492. which I think is the latest prod version. So I suspect that at some point  the CLI project started using an updated version version of okhttp. 

In summary in our scenario 
sonar-scanner-cli-3.2.0.1227 - WORKS with certs that do not have a SubjectAlternativeName
sonar-scanner-cli-3.3.0.1492 - WIILL NOT WORK without SubjectAlternativeName

After we regenerated our certs properly the problem went away 

hope this helps
max

**Comment by gfvirga** [comment]
Max,
This was very helpful. 
Thank you!

**Comment by henryju** [comment]
@MMirabito thanks for having confirmed the issue, it will be helpful to others I guess. Also I suggest you keep using the latest official release. The build I sent to you is not production ready, it was just a try to help investigation.

**Comment by bellingard** [comment]
Thanks @henryju for the investigation. I'm closing this issue since I don't believe there's a problem on the scanner for NPM side.

**Comment by eric-gagnon** [comment]
Thanks for the info on "subjectAltNames". It does look like it's a dead end for me. I was trying to run sonar-scanner from the cloud and send the results to our private SonarQube instance which is behind a Web application firewall. The certificate that is used currently is the one from the waf (ssl offloading) so no way I can add he required subjectAltNames.

Would gladly take any alternatives that could still be safe...
